src/platform/forms-system/src/js/fields/FileField.jsx

```diff
@@ -147,6 +305,22 @@ export default class FileField extends React.Component {
                 errorSchema,
               );
               const attachmentNameErrors = _.get([index, 'name'], errorSchema);
+              const showPasswordInput =
+                pdfPasswordFeature && // feature flag
+                (files[index].isEncrypted || // needs password
+                  // incorrect password, returning error message
+                  (files[index].password && files[index].errorMessage));
+              const showPasswordSuccess =
+                pdfPasswordFeature &&
+                file.password &&
+                !hasErrors &&
+                !file.errorMessage;
+
+              if (showPasswordInput) {
+                setTimeout(() => {
+                  focusElement(`[name="get_password_${index}"]`);
+                }, 100);
+              }
```

Comment:
This seems useful, but it is a pattern that I'm worried could cause unexpected behavior if the rendering takes longer than expected, and it seems like it could be tricky to enforce if we wanted to test that a certain element grabbed focus when we enter an error state. I'm not suggesting anything right now, I'm just calling it out so I can see what others have to say about it.

Reply:
I thought about applying the focus to the element within the `ShowPdfPassword` component when the component mounted, but then it might be stealing focus from another password input, if more than one is open at a time. I don't know a good solution either.